ddtrace/tracer/spancontext.go

```diff
@@ -345,6 +345,18 @@ func (t *trace) setSamplingPriorityLocked(p int, sampler samplernames.SamplerNam
 	return updatedPriority
 }
 
+func (t *trace) isTraceLocked() bool {
+	t.mu.RLock()
+	defer t.mu.RUnlock()
+	return t.locked
+}
+
+func (t *trace) setTraceLocked(locked bool) {
```

Comment:
nit (optional, you can ignore this): I would suggest removing the "trace" part in the functions as [it introduces repetition](https://go.dev/blog/package-names#naming-package-contents).

```suggestion
func (t *trace) isLocked() bool {
	t.mu.RLock()
	defer t.mu.RUnlock()
	return t.locked
}

func (t *trace) setLock(locked bool) {
```

Reply:
applied